### PR TITLE
Change project type API unit tests

### PR DESCRIPTION
**Pull request by Pezmc**
Unit tests of the API changes for #1610.

Best viewed with [?w=1](https://github.com/flowforge/flowforge/pull/1631/files?w=1) due to indent changes in the tests.

Pre-cursor to https://github.com/flowforge/flowforge/pull/1623

### CODE REVIEW

File: test/unit/forge/routes/api/project_spec.js [hunk 1]
```diff
@@ -755,222 +755,410 @@ describe('Project API', function () {
...
-                TestObjects.tokens.project,
-                [{ id: 'node1' }],
-                { testCreds: 'abc' },
-                'key1',
-                {
-                    httpAdminRoot: '/test-red',
-                    env: [
-                        { name: 'one', value: 'a' },
-                        { name: 'two', value: 'b' }
+                response.statusCode.should.equal(200)
```
Comment (Pezmc) [inline test/unit/forge/routes/api/project_spec.js:794]: Unfortunately, I couldn't find a way to assert the project is suspended as the request only returns once the project has been restarted.

**Review by knolleary — APPROVED** [review]

File: test/unit/forge/routes/api/project_spec.js (hunk 1 above)
Comment (knolleary) [reply test/unit/forge/routes/api/project_spec.js:794]: One approach would be to add some sort of internal logging to the stub driver that can be used to check what has been done to a given project. But not sure that's worth the effort at this point.

Comment (Pezmc) [reply test/unit/forge/routes/api/project_spec.js:794]: We could spy on the audit logs actually.

Comment (knolleary) [reply test/unit/forge/routes/api/project_spec.js:794]: I did wonder that, but couldn't recall if the audit logging is done at that level.

**Pull request merged**